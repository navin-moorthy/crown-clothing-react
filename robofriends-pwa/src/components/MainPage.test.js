import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

let MainPageComponent;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false
  };

  MainPageComponent = shallow(<MainPage {...mockProps} />);
});

it("expect to render MainPage component", () => {
  expect(MainPageComponent.debug()).toMatchSnapshot();
});

it("filter robots correctly", () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: "1",
        name: "John",
        email: "john@email.com"
      }
    ],
    searchField: "john",
    isPending: false
  };
  const MainPageComponent2 = shallow(<MainPage {...mockProps2} />);
  expect(MainPageComponent2.instance().filteredRobots()).toEqual([
    {
      id: "1",
      name: "John",
      email: "john@email.com"
    }
  ]);
});

it("filter robots correctly 2", () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: "1",
        name: "John",
        email: "john@email.com"
      }
    ],
    searchField: "a",
    isPending: false
  };

  const filteredRobots = [];
  const MainPageComponent3 = shallow(<MainPage {...mockProps3} />);
  expect(MainPageComponent3.instance().filteredRobots()).toEqual(
    filteredRobots
  );
});
