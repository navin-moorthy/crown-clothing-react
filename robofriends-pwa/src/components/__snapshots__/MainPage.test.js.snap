// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundary>
      <CardList robots={{...}} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;
